src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_ARRAY_LENGTH = 10;
+    private int size;
+    private T[] objects;
+
+    public ArrayList() {
+        objects = (T[]) new Object[START_ARRAY_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        doNeedGrowth();
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Such index does not exist: " + index);
+        }
+        doNeedGrowth();
+        System.arraycopy(objects, index, objects, index + 1, size - index);
+        objects[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validate(index);
+        return objects[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validate(index);
+        objects[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validate(index);
+        T removed = objects[index];
+        System.arraycopy(objects, index + 1, objects, index, size - 1 - index);
+        objects[--size] = null;
+        return removed;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        size--;
+        for (int j, i = 0; i < size; i++) {
```

Comment:
what is 'j' ?

Reply:
I was trying to do this fast, I just didn`t notice..
Sorry, next time I will be more careful